packages/muelu/test/maxwell/Maxwell3D.cpp

```diff
@@ -432,7 +430,7 @@ int MainWrappers<double,LocalOrdinal,GlobalOrdinal,Node>::main_(Teuchos::Command
     tm = Teuchos::null;
 
     if (solverName == "Belos") {
-      tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("Maxwell: 2 - Build Belos solver etc")));
+      auto tm2 = TimeMonitor::getNewTimer("Maxwell: 2 - Build Belos solver etc");
```

Comment:
What does this actually do?  Does this next timers or no?

Reply:
@DrBooom explained that this fixed "segfault on exit/return" issues, likely by ensuring that TimeMonitor instances don't persist beyond where they should.

If we think this is broken, let's all get together in a hackathon and fix it together and push it.

Reply:
In the original code, each timer timed a disjoint region (modulo the usual issues).  These timers  *look* like they're overlapping.  That isn't correct behavior.

Reply:
@csiefer2 OK, I agree that this actually would change timer scopes.  This could be relevant to @DrBooom 's timings, though I think he collected timings before making this change.  @DrBooom had to do _something_ here, else the code would crash.  I propose that for this code _only_, we fix it to use TimeMonitor in the correct way: for example:
```c++
{ 
  TimeMonitor tm1 (...);
  thing_to_time_1 ();
}
{
  TimeMonitor tm2 (...);
  thing_to_time_2 ();
}
```
etc.  Does that sound like a good plan?  @DrBooom I will be happy to help you with this.

Reply:
I've fixed this issue by simply assigning the old timer to null before the creation of the new timer. Is that acceptable?